src/ucc/api/ucc.h

```diff
@@ -909,6 +909,27 @@ ucc_status_t ucc_context_destroy(ucc_context_h context);
 ucc_status_t ucc_context_get_attr(ucc_context_h context,
                                   ucc_context_attr_t *context_attr);
 
+/**
+ *  @ingroup UCC_CONTEXT
+ *
+ *  @brief The routine release the resources associated with the
+ *         context attributes
+ *
+ *  @param [out]  context_attr     Attributes of the communication context
+ *
+ *  @parblock
+ *
+ *  @b Description
+ *
+ *  @brief @ref ucc_context_free_attr routine releases the resources
+ *  associated with the context attributes described by @ref ucc_context_attr.
+ *
+ *  @endparblock
+ *
+ *  @return Error code as defined by ucc_status_t
+ */
+
+ucc_status_t ucc_context_free_attr(ucc_context_attr_t *context_attr);
```

Comment:
Do we need this? The context attributes are created by the library and it should destroy it.

Reply:
But when? For example, if user asks for context address, then there is memory allocated during ucc_context_get_attr call and that stores the address. User can take the buffer with the address from attribs. When library can clear it? User needs another api to release any resources associated with attribs (in the case get_attr implies resource allocation like in our case)

Reply:
Why not during the context destroy? If we provide this free attributes approach, we need to make sure that free attributes is called before the context destroy.

Reply:
agree - this makes sense, did think about it. I made the change: ctx address is packed on demand and cached on context. Released in context_destroy